### DISCUSSION TITLE
New Collection review: netapp.azure

### DISCUSSION THREAD

**Discussion by carchi8py** [ansible-collections/ansible-inclusion#17] [Resolved reviews]
We've created a few new collections for our other products (same team that currently support the existing NetApp collections). We do monthly update to each collection (And bug fixes through out the month if needed)

- Ansible Galaxy: https://galaxy.ansible.com/netapp/azure
- Github: https://github.com/ansible-collections/netapp
- Issues Tracker: https://github.com/ansible-collections/netapp/issues
- Github Handles:  @carchi8py @lonico

**Comment by felixfontein** [comment]
Most things have already been covered in the other two collections. I'll add here some points that are specific to this one:
1. https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/azure/galaxy.yml#L10 must have minimum requirement `>=1.0.0`.
2. https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/azure/plugins/doc_fragments/netapp.py#L14 is invalid.
3. There is a collection build artefact (https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/azure/netapp-azure-20.8.0.tar.gz) inside the repository.

**Comment by dmsimard** [comment]
# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

**Public availability and communication:**
- [X] published on Ansible Galaxy [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [X] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
  Was recently added via https://github.com/ansible-collections/netapp/issues/93
- [ ] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
  None found, this is required and has been pointed out in other netapp reviews such as this one: https://github.com/ansible-collections/ansible-inclusion/discussions/16#discussioncomment-497501
- [X] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [ ] have a public SCM repository, releases of the collection are tagged in this repository
  This is complicated because multiple collections share the same repository and tags seem to apply to every collection.

**Standards and documentation:**
- [ ] adheres to [semantic versioning](https://semver.org/)
  Semantic versioning is important for inclusion in the package so that we may properly pin versions and control major/breaking changes as well as forward/backwards compatibility between versions of Ansible. This has already been pointed out in another review: https://github.com/ansible-collections/ansible-inclusion/discussions/16#discussioncomment-506143
- [ ] follows licensing rules
  Plugins are GPLv3 but galaxy.yml mentions GPLv2. Already pointed out in another review: https://github.com/ansible-collections/ansible-inclusion/discussions/18#discussioncomment-519785
- [ ] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
- [X] only uses allowed plugin types in the `plugins/` directory
- [X] has `README.md`
- [X] documentation, examples, and return use FQCNs for `M(..)`, examples, and `seealso` subsections
- [X] modules (or plugins) from ansible-base use `ansible.builtin.` as a FQCN prefix
- [X] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [X] FQCNs is used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-base

**Collection management:**
- [X] `galaxy.yml` having `tags` field set
- [ ] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
  Requirement on ``azure.azcollection`` in galaxy.yml is set to ``*``, it should be ``>=1.0.0``.
- [ ] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
  This is missing.
- [X] has changelog, preferably with `changelogs/changelog.yaml`

**Tests:**
- [X] passed `ansible-test sanity`
  Although a subset is likely not run from looking at the results of a PR: https://github.com/ansible-collections/netapp/pull/87
- [X] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [ ] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [X] all CI tests run against every pull request
- [ ] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [ ] all CI tests run regularly (nightly, or at least once per week)

**Comment by tadeboro** [comment]
Just a reminder: collections that want to be part of the Ansible 4.0.0 package need to get approval by 2021-04-26 ([full schedule](https://docs.ansible.com/ansible/devel/roadmap/COLLECTIONS_4.html#release-schedule)). This inclusion request still has some open issues that need to get addressed by then.

**Reply by tadeboro** [reply]
**Last reminder for inclusion submission**

Collections that want to be part of the Ansible 4.0.0 package need to get approval by 2021-04-26, which is just one week away ([full schedule](https://docs.ansible.com/ansible/devel/roadmap/COLLECTIONS_4.html#release-schedule)). Make sure you submit your collection for the final review a few days before the deadline (reviews take a non-trivial amount of time) and that your collection does meet all criteria listed in the [requirements document](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst).

**Comment by tadeboro** [comment]
Thank you for your submission, @carchi8py.

The items in the checklist can be in one of the following states:

  * Items with a checkmark and no attached comment are OK and need no intervention from your side.
  * Items with a checkmark that contain a comment will not block the inclusion process, but by following the advice given in the comment, the collection's quality can be improved.
  * Items with no checkmark and no extra details are waiting for evaluation.
  * Items with no checkmark that contain a comment need your attention.

Feel free to ask any questions and provide feedback. Thanks in advance for your understanding.

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

**Public availability and communication:**
- [x] published on Ansible Galaxy [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
      **Comment:** The netapp documentation fragment is invalid: https://github.com/ansible-collections/netapp/blob/5b7c1f20dfc383c85f6d44c578670b7fe27eca05/ansible_collections/netapp/azure/plugins/doc_fragments/netapp.py#L14, but is not used anywhere. This will not block the inclusion, but should be fixed/simplified in the future.
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`

**Tests:**
- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)

**Reply by tadeboro** [reply]
I did an initial review of the collection, and things are generally in good shape. A few issues need to be resolved before this collection can become part of the community Ansible package, but those should be relatively straightforward to resolve.

**Reply by lonico** [reply]
I'm addressing the review comments for CoC, FCQNs, azcollection and ansible minimum versions.  I think @carchi8py already addressed the CI issues with CloudManager.   The README.md is already there, but I'll update it for the CoC. 

Sanity tests: https://github.com/ansible-collections/netapp/runs/2383401825

**Reply by tadeboro** [reply]
@lonico Yes, the CI stuff is fixed since all of the netapp.* collections share the same CI pipeline.

**Reply by lonico** [reply]
changes pushed to GitHub.

**Reply by tadeboro** [reply]
Thank you, @carchi8py, for your perseverance. Once the next release hits Ansible Galaxy, this collection is ready for inclusion as far as I am concerned. Congratulations!

**Reply by carchi8py** [reply]
@tadeboro the changes above have been uploaded in to latest release of azure https://galaxy.ansible.com/netapp/azure

**Reply by felixfontein** [reply]
I agree with @tadeboro.

**Comment by felixfontein** [comment]
This collection has been prepared for inclusion in Ansible 4 in this PR: https://github.com/ansible-community/ansible-build-data/pull/66

**Comment by tadeboro** [comment]
The `netapp.azure` Ansible Collection was accepted into the Ansible community package. The relevant part of the community meeting starts [here](https://meetbot.fedoraproject.org/ansible-community/2021-04-28/ansible_community_meeting.2021-04-28-18.00.log.html#l-163).

Congratulations!
